db/migrate/20171012202650_cleanup_invalid_contacts.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class CleanupInvalidContacts < ActiveRecord::Migration[5.1]
+  class Contact < ApplicationRecord
+    belongs_to :user
+    belongs_to :person
+
+    has_many :aspect_memberships, dependent: :destroy
+    has_many :aspects, through: :aspect_memberships
+
+    before_destroy :destroy_notifications
+
+    def destroy_notifications
+      Notification.where(
+        target_type:  "Person",
+        target_id:    person_id,
+        recipient_id: user_id,
+        type:         "Notifications::StartedSharing"
+      ).destroy_all
+    end
+  end
+
+  class User < ApplicationRecord
+    has_one :person, inverse_of: :owner, foreign_key: :owner_id
+  end
+
+  class Notification < ApplicationRecord
+    has_many :notification_actors, dependent: :destroy
```

Comment:
Since `NotificationActor` doesn't do anything special on destroy, you can use `dependent: :delete_all` here, which is faster, because it doesn't instantiate the model (we could change that in the real model too).

Reply:
Maybe we can use `:delete_all` in `Contact` `:aspect_memberships` relation in migration model as well?

Reply:
Sure, that makes sense 👍